jfr-connection/src/main/java/io/opentelemetry/contrib/jfr/connection/FlightRecorderDiagnosticCommandConnection.java

```diff
@@ -197,41 +224,41 @@ public void closeRecording(long id) {
-  static boolean jdkHasUnlockCommercialFeatures(MBeanServerConnection mBeanServerConnection) {
-    try {
-      RuntimeMXBean runtimeMxBean =
-          ManagementFactory.getPlatformMXBean(mBeanServerConnection, RuntimeMXBean.class);
-      String javaVmVendor = runtimeMxBean.getVmVendor();
-      String javaVersion = runtimeMxBean.getVmVersion();
-      return javaVmVendor.contains("Oracle Corporation")
-          && javaVersion.matches("(?:^1\\.8|9|10).*");
-    } catch (IOException e) {
-      return false;
-    }
-  }
-
   // visible for testing
   static void assertCommercialFeaturesUnlocked(
       MBeanServerConnection mBeanServerConnection, ObjectName objectName)
       throws IOException, JfrConnectionException {
 
     try {
-      Object unlockedMessage =
-          mBeanServerConnection.invoke(objectName, "vmCheckCommercialFeatures", null, null);
-      if (unlockedMessage instanceof String) {
-        boolean unlocked = ((String) unlockedMessage).contains("unlocked");
-        if (!unlocked) {
-          throw JfrConnectionException.canonicalJfrConnectionException(
-              FlightRecorderDiagnosticCommandConnection.class,
-              "assertCommercialFeaturesUnlocked",
-              new UnsupportedOperationException(
-                  "Unlocking commercial features may be required. This must be explicitly enabled by adding -XX:+UnlockCommercialFeatures"));
-        }
+      Object[] params = new Object[] {new String[] {}};
+      MBeanInfo mBeanInfo = mBeanServerConnection.getMBeanInfo(objectName);
+      if (mBeanInfo == null) {
+        throw JfrConnectionException.canonicalJfrConnectionException(
+            FlightRecorderDiagnosticCommandConnection.class,
+            "assertCommercialFeaturesUnlocked",
+            new NullPointerException("Could not get MBeanInfo for " + objectName));
       }
-    } catch (InstanceNotFoundException | MBeanException | ReflectionException ignored) {
-      // If the MBean doesn't have the vmCheckCommercialFeatures method, then we can't check it.
+      Optional<MBeanOperationInfo> operation =
+          Arrays.stream(mBeanInfo.getOperations())
+              .filter(it -> "vmUnlockCommercialFeatures".equals(it.getName()))
+              .findFirst();
+
+      if (operation.isPresent()) {
+        mBeanServerConnection.invoke(objectName, "vmUnlockCommercialFeatures", params, signature);
+      } else {
+        // No vmUnlockCommercialFeatures command.
+        return;
+      }
+    } catch (InstanceNotFoundException
+        | IntrospectionException
+        | MBeanException
+        | ReflectionException e) {
+      throw JfrConnectionException.canonicalJfrConnectionException(
+          FlightRecorderDiagnosticCommandConnection.class,
+          "assertCommercialFeaturesUnlocked",
+          new UnsupportedOperationException(
+              "Unlocking commercial features may be required. This must be explicitly enabled by adding -XX:+UnlockCommercialFeatures",
```

Comment:
My understanding is that the method unlocks the commercial features with the help of an MBean operation. In this case, is the `-XX:+UnlockCommercialFeatures` message relevant?